### ISSUE TITLE
flutter build ios-framework does not generate obfuscation symbols

### ISSUE THREAD

**Issue by Timoteohss** [flutter/flutter#64730]
## Steps to Reproduce

1. Run `flutter build ios-framework --no-debug --no-profile --xcframework --no-universal --cocoapods --obfuscate --split-debug-info=<a_path>`.

**Expected results:** 

<a_path> containing ios-framework symbol map files

**Actual results:** 

<a_path> is empty

<details>
  <summary>Logs</summary>

```
Building frameworks for com.[app] in release mode...
 ├─Populating Flutter.framework...                                  3,4s
 ├─Building App.framework...                                       24,7s
 ├─Building plugins...                                             71,3s
 └─Moving to build/ios/framework/Release                            0,1s
Frameworks written to
/Users/timoteoholanda/repos/[app]/build/ios/framework.
```

```
Analyzing next_flutter...

   info • Unused import: 'package:flutter/material.dart' •
          test/widget_test.dart:8:8 • unused_import
   info • Unused import: 'package:[app]/main.dart' •
          test/widget_test.dart:10:8 • unused_import

2 issues found. (ran in 6.6s)
```

```
[✓] Flutter (Channel stable, 1.20.2, on Mac OS X 10.15.5 19F96, locale pt-BR)
    • Flutter version 1.20.2 at /Users/timoteoholanda/development/flutter
    • Framework revision bbfbf1770c (2 weeks ago), 2020-08-13 08:33:09 -0700
    • Engine revision 9d5b21729f
    • Dart version 2.9.1


[✓] Android toolchain - develop for Android devices (Android SDK version 29.0.3)
    • Android SDK at /Users/timoteoholanda/Library/Android/sdk
    • Platform android-30, build-tools 29.0.3
    • Java binary at: /Applications/Android
      Studio.app/Contents/jre/jdk/Contents/Home/bin/java
    • Java version OpenJDK Runtime Environment (build
      1.8.0_212-release-1586-b4-5784211)
    • All Android licenses accepted.

[✓] Xcode - develop for iOS and macOS (Xcode 11.1)
    • Xcode at /Applications/Xcode.app/Contents/Developer
    • Xcode 11.1, Build version 11A1027
    • CocoaPods version 1.9.2

[✓] Android Studio (version 3.6)
    • Android Studio at /Applications/Android Studio.app/Contents
    • Flutter plugin version 45.1.1
    • Dart plugin version 192.8052
    • Java version OpenJDK Runtime Environment (build
      1.8.0_212-release-1586-b4-5784211)

[✓] VS Code (version 1.48.2)
    • VS Code at /Applications/Visual Studio Code.app/Contents
    • Flutter extension version 3.13.2

[✓] Connected device (1 available)

• No issues found!
```

</details>

Android builds using `flutter build aar --obfuscate --split-debug-info=<a_path>` does work as intended, but ios-frameworks is not.

**Comment by TahaTesser** [comment]
I can reproduce the issue on `stable` & `master`
[logs.txt](https://github.com/flutter/flutter/files/5141282/logs.txt)

<details> 
<summary>flutter doctor -v</summary> 

```
[✓] Flutter (Channel stable, 1.20.2, on Mac OS X 10.15.6 19G2021, locale en-GB)
    • Flutter version 1.20.2 at /Users/tahatesser/Code/flutter_stable
    • Framework revision bbfbf1770c (2 weeks ago), 2020-08-13 08:33:09 -0700
    • Engine revision 9d5b21729f
    • Dart version 2.9.1

 
[✓] Android toolchain - develop for Android devices (Android SDK version 30.0.1)
    • Android SDK at /Users/tahatesser/Code/sdk
    • Platform android-30, build-tools 30.0.1
    • ANDROID_HOME = /Users/tahatesser/Code/sdk
    • Java binary at: /Applications/Android Studio.app/Contents/jre/jdk/Contents/Home/bin/java
    • Java version OpenJDK Runtime Environment (build 1.8.0_242-release-1644-b3-6222593)
    • All Android licenses accepted.

[✓] Xcode - develop for iOS and macOS (Xcode 11.6)
    • Xcode at /Applications/Xcode.app/Contents/Developer
    • Xcode 11.6, Build version 11E708
    • CocoaPods version 1.9.3

[✓] Android Studio (version 4.0)
    • Android Studio at /Applications/Android Studio.app/Contents
    • Flutter plugin version 48.1.2
    • Dart plugin version 193.7547
    • Java version OpenJDK Runtime Environment (build 1.8.0_242-release-1644-b3-6222593)

[✓] VS Code (version 1.48.2)
    • VS Code at /Applications/Visual Studio Code.app/Contents
    • Flutter extension version 3.13.2

[✓] Connected device (1 available)
    • iPhone 11 (mobile) • A63D6F51-59F3-4A75-961F-3769709F3A92 • ios •
      com.apple.CoreSimulator.SimRuntime.iOS-13-6 (simulator)

• No issues found!
```

</details> 


<details> 
<summary>flutter doctor -v</summary> 

```
[✓] Flutter (Channel master, 1.22.0-2.0.pre.105, on Mac OS X 10.15.6 19G2021, locale en-GB)
    • Flutter version 1.22.0-2.0.pre.105 at /Users/tahatesser/Code/flutter_master
    • Framework revision cad4d1333e (9 hours ago), 2020-08-27 17:48:05 -0700
    • Engine revision 77dd1c05b9
    • Dart version 2.10.0 (build 2.10.0-67.0.dev)

[✓] Android toolchain - develop for Android devices (Android SDK version 30.0.1)
    • Android SDK at /Users/tahatesser/Code/sdk
    • Platform android-30, build-tools 30.0.1
    • ANDROID_HOME = /Users/tahatesser/Code/sdk
    • Java binary at: /Applications/Android Studio.app/Contents/jre/jdk/Contents/Home/bin/java
    • Java version OpenJDK Runtime Environment (build 1.8.0_242-release-1644-b3-6222593)
    • All Android licenses accepted.

[✓] Xcode - develop for iOS and macOS (Xcode 11.6)
    • Xcode at /Applications/Xcode.app/Contents/Developer
    • Xcode 11.6, Build version 11E708
    • CocoaPods version 1.9.3

[✓] Chrome - develop for the web
    • Chrome at /Applications/Google Chrome.app/Contents/MacOS/Google Chrome

[✓] Android Studio (version 4.0)
    • Android Studio at /Applications/Android Studio.app/Contents
    • Flutter plugin version 48.1.2
    • Dart plugin version 193.7547
    • Java version OpenJDK Runtime Environment (build 1.8.0_242-release-1644-b3-6222593)

[✓] VS Code (version 1.48.2)
    • VS Code at /Applications/Visual Studio Code.app/Contents
    • Flutter extension version 3.13.2

 
[✓] Connected device (4 available)            
    • iPhone 11 (mobile) • A63D6F51-59F3-4A75-961F-3769709F3A92 • ios            •
      com.apple.CoreSimulator.SimRuntime.iOS-13-6 (simulator)
    • macOS (desktop)    • macos                                • darwin-x64     • Mac OS X 10.15.6
      19G2021
    • Web Server (web)   • web-server                           • web-javascript • Flutter Tools
    • Chrome (web)       • chrome                               • web-javascript • Google Chrome
      84.0.4147.135

• No issues found!
```

</details>

**Comment by jmagman** [comment]
I thought this was hooked up in https://github.com/flutter/flutter/pull/59874 and https://github.com/flutter/flutter/pull/60041, \cc @jonahwilliams

**Comment by toneyzeng** [comment]
I have the same problem in v1.22.

**Comment by efan006** [comment]
I have the same problem in Flutter 2.2.1 stable.
And It seems didn't obfuscate anything because two "App.framework"s builded with/without flags "--obfuscate --split-debug-info=<a_path>" are exactly the same size. and there is only a few differences in binary comparing.

**Comment by jmagman** [comment]
> I have the same problem in Flutter 2.2.1 stable.
> And It seems didn't obfuscate anything because two "App.framework"s builded with/without flags "--obfuscate --split-debug-info=<a_path>" are exactly the same size. and there is only a few differences in binary comparing.

The fix is in 2.3.0-12.0.pre.

**Comment by efan006** [comment]
> > I have the same problem in Flutter 2.2.1 stable.
> > And It seems didn't obfuscate anything because two "App.framework"s builded with/without flags "--obfuscate --split-debug-info=<a_path>" are exactly the same size. and there is only a few differences in binary comparing.
> 
> The fix is in 2.3.0-12.0.pre.

Thanks, it works when I upgrade to 2.3.0-16.0.pre
